Lib/inspect.py

```diff
@@ -186,6 +187,12 @@ def ismethod(object):
     """Return true if the object is an instance method."""
     return isinstance(object, types.MethodType)
 
+def ispackage(object):
+    """Return true if the object is a package."""
+    if ismodule(object) and object.__spec__ is not None:
+        return object.__spec__.parent == object.__spec__.name
+    return False
```

Comment:
I'm not an expert on importlib, but wouldn't it be enough to check if an object has a `__path__` attribute?

Reply:
Yes! That's true. I complicated the problem before.

But I still think it is necessary to judge whether the object is a module or not, according to its [exact definition](https://docs.python.org/3/reference/import.html#path-attributes-on-modules).